patching/plugin-commands-patching/src/patchCommit.ts

```diff
@@ -129,3 +133,34 @@ function removeTrailingAndLeadingSlash (p: string) {
   }
   return p
 }
+async function filterAndCopyFiles (source: string, destination: string) {
+  const files = await packlist({ path: source })
+  const shouldCopyFiles = await checkIfAllFilesExist(files, source)
+
+  if (!shouldCopyFiles) {
+    return source
+  }
+
+  await Promise.all(
+    files.map(async (file) => {
+      const sourcePath = path.join(source, file)
+      const destinationPath = path.join(destination, file)
+      const destDir = path.dirname(destinationPath)
+      await fs.promises.mkdir(destDir, { recursive: true })
+      await fs.promises.copyFile(sourcePath, destinationPath)
+    })
+  )
+
+  return destination
+}
+async function checkIfAllFilesExist (files: string[], basePath: string) {
```

Comment:
My version of eslint seems to be inconsistent with the main branch eslint, I have modified it according to the latest eslint
```ts
async function checkIfAllFilesExist (files: string[], basePath: string) {
  const promises = files.map(async (file) => {
    const filePath = path.join(basePath, file)
    try {
      await fs.promises.access(filePath)
    } catch (error) {
      return false
    }
    return true // File exists
  })

  const results = await Promise.all(promises)
  return results.every((result) => result) // Check if all files exist
}
```